### PR TITLE
Create a pseudo-stacking context for positioned floats.

### LINKED ISSUE #7828: stacking issue with float and position absolute
```html
<style>
  div {
    width: 100px;
    height: 100px;
  }

  #a {
    position: absolute;
    top: 20px;
    left: 20px;
    background: red;
  }

  #b {
    float: left;
    position: relative;
    background: green;
  }
</style>

<body>
  <div id="a"></div>
  <div id="b"></div>
</body>
```

screen shot 2015-10-02 at 11 47 10

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #7828 

That was surprisingly easy...

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7834)

### CODE REVIEW

File: components/layout/display_list_builder.rs [hunk 1]
```diff
@@ -1717,6 +1717,12 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
                                                       ScrollPolicy::Scrollable,
                                                       StackingContextCreationMode::Normal))
         } else {
+            match self.fragment.style.get_box().position {
+                position::T::static_ => {}
+                _ => {
+                    display_list.form_pseudo_stacking_context_for_positioned_content();
+                }
+            }
```
Comment (pcwalton) [inline components/layout/display_list_builder.rs]: nit: use `if ....position != position::T::static_ { ... }` maybe to save a level of indentation?

**Comment by jdm** [comment]
This may be a legit new test pass; take a look locally please?

**Comment by notriddle** [comment]
Sure.

**Comment by notriddle** [comment]
Looks right to me.

**Pull request merged**